README.md

```diff
@@ -84,6 +84,74 @@ is uploaded to S3 as an asset _as is_, still encoded. The custom
 resource Lambda then decodes the secrets (in memory, never on disk)
 and puts them into the SecretsManager secret.
 
+## KMS Key Policy Requirements
+
+For the Lambda function to successfully decrypt SOPS files, the KMS key used for encryption must have a key policy that allows the Lambda execution role to perform decryption operations. While this CDK construct grants the Lambda broad KMS permissions via IAM policies (`kms:*`), KMS key policies are resource-based policies that can override IAM permissions.
+
+Both the IAM policy (granted by this construct) AND the KMS key policy must allow the Lambda execution role to use the key.
+
+### Required KMS Key Policy Permissions
+
+The KMS key policy must allow the Lambda execution role to perform the following actions:
+
+- `kms:Decrypt` - Required to decrypt the SOPS file
+- `kms:DescribeKey` - May be required for key metadata operations
```

Comment:
Make it clearer that this is the minimum permissions that should be granted for this construct to function correctly.